modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/sorted/io/SortedIndexTreeIo.java

```diff
@@ -117,39 +166,48 @@ default int compare(
             int idx,
             SortedIndexRowKey rowKey
     ) throws IgniteInternalCheckedException {
-        int off = offset(idx);
+        final int off = offset(idx);
+
+        int indexColumnsSize = getShort(pageAddr + off, SIZE_OFFSET);
 
-        long link = readPartitionless(partitionId, pageAddr, off + INDEX_COLUMNS_LINK_OFFSET);
+        ByteBuffer firstBinaryTupleBuffer;
 
-        //TODO Add in-place compare in IGNITE-17671
-        ReadIndexColumnsValue indexColumnsTraversal = new ReadIndexColumnsValue();
+        if (indexColumnsSize != NOT_FULLY_INLINE) {
```

Comment:
This line is very hard to read. Double negation should be avoided at all cost. You could have swapped two branches, for example, or created a method with a good name

Reply:
Fix it